src/app/components/todolist/todolist.component.html

```diff
@@ -5,26 +5,27 @@
 <!--ngFor loops over each of the todo items in the list so that they're all displayed in the same format-->
 <!--added a model to the input for the todo list.  Therefore we save a property to the todo object of Title-->
 <!--added an event listener on the enter key, which calls the addTodo method-->
+<!--added ngmodel todo.completed to the checkbox which sets the boolean flag completed to true against the checkbox-->
 <div class="todo-item" *ngFor="let todo of todos">
   <div class="todo-item-left">
-    <input type="checkbox">
-    <div class="todo-item-label">{{ todo.title }}</div>
+    <input type="checkbox" [(ngModel)]="todo.completed">
+    <div class="todo-item-label" [ngClass]="{ 'completed' : todo.completed }">{{ todo.title }}</div>
   </div>
   <div class="remove-item" (click)="deleteTodo(todo.id)">
     &times;
   </div>
 </div>
 
 <div class="extra-container">
-  <div><label><input type="checkbox"> Check All</label></div>
-  <div>n items left</div>
+  <div><label><input type="checkbox" (change)="checkAllTodos()"> Check All</label></div>
+  <div>{{ todoRemaining() }} items left</div>
```

Comment:
This does work, but keep in mind when calling a method from the HTML that it will be called multiple times from the angular lifecycle hooks. In this case the method does a small amount of compute, but where you might be doing some intensive operations, then you will see a performance issue.

Take a read of this: https://blog.appverse.io/why-it-is-a-bad-idea-to-use-methods-in-the-html-templates-with-angular-2-30d49f0d3b16

That article take it to the extreme, but you hopefully understand what I mean

Reply:
I'll give the article a read, thanks for the feedback and good to know.  I imagine with larger apps they can drag down the site a bit.